### PR TITLE
fix #2114: make doc subsections header bold

### LINKED ISSUE #2114: Improve doc examples rendering
Currently we are rendering our examples in the docs like this https://pytorch.org/ignite/generated/ignite.engine.engine.Engine.html#ignite.engine.engine.Engine which is not very neat.
What we want to do is to find a better way to make it more good looking, and update our docs examples

image

- Replace "Examples" with some sphinx tag or something else, for example, like scikit-image does:

image

### PR DESCRIPTION
**Pull request by toxa23**
Fixes #2114

Description:
part 1 of 7, adding CSS style for docstring section headers

Check list:

- [ ] New tests are added (if a new feature is added)
- [ ] New doc strings: description and/or example code are in RST format
- [ ] Documentation is updated (if required)

### CODE REVIEW

**Comment by vfdev-5** [comment]
@ydcjeff please help with reviewing this PR. Thanks!

**Review by vfdev-5 — APPROVED** [review]
Thanks for the PR @toxa23 
I approve as it looks good to me. Let's see what @ydcjeff says

File: docs/source/_templates/_static/css/ignite_theme.css [hunk 1]
```diff
@@ -209,3 +209,14 @@ so make sure not to go above that value */
...
+article.pytorch-article p.rubric {
+    font-weight: bold;
+    font-size: 19px;
+    margin: 15px 0 10px 0;
+}
+
+pre {
+    border: 1px solid rgba(0,0,0,0.15);
+    border-radius: 4px;
+}
```
Comment (ydcjeff) [inline docs/source/_templates/_static/css/ignite_theme.css:222]:
```suggestion
article.pytorch-article p.rubric {
  font-weight: bold;
  font-size: 1.25rem;
  margin: 15px 0 10px 0;
}

.highlight {
  border: 1px solid rgba(0,0,0,0.15);
  border-radius: 4px;
}
```

**Review by ydcjeff — APPROVED** [review]
LGTM

**Pull request merged**